test/common/alerts_firing.go

```diff
@@ -61,10 +76,99 @@ func (e *alertsFiringError) isValid() bool {
 	return !e.deadMansSwitchFiring || len(e.alertsFiring) != 0 || len(e.alertsPending) != 0
 }
 
-// TestIntegreatlyAlertsFiring reports any firing or pending alerts
 func TestIntegreatlyAlertsFiring(t *testing.T, ctx *TestingContext) {
 	var lastError error
 
+	if newErr := getFiringAlerts(t, ctx); newErr != nil {
+		lastError = newErr
+		podLogs(t, ctx)
+		t.Fatal(lastError.Error())
+	}
+
+}
+
+func getFiringAlerts(t *testing.T, ctx *TestingContext) error {
+	output, err := execToPod("curl localhost:9090/api/v1/alerts",
+		"prometheus-application-monitoring-0",
+		MonitoringOperatorNamespace,
+		"prometheus",
+		ctx)
+	if err != nil {
+		return fmt.Errorf("failed to exec to prometheus pod: %w", err)
+	}
+
+	// get all found rules from the prometheus api
+	var promApiCallOutput prometheusAPIResponse
+	err = json.Unmarshal([]byte(output), &promApiCallOutput)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal json: %w", err)
+	}
+	var alertsResult prometheusv1.AlertsResult
+	err = json.Unmarshal(promApiCallOutput.Data, &alertsResult)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal json: %w", err)
+	}
+
+	// create a custom alerts error to keep track of all pending and firing alerts
+	alertsError := &alertsFiringError{
+		alertsFiring:         []alertTestMetadata{},
+		deadMansSwitchFiring: true,
+	}
+
+	// // check if any alerts other than DeadMansSwitch are firing
+	for _, alert := range alertsResult.Alerts {
+		alertName := string(alert.Labels["alertname"])
+		alertMetadata := alertTestMetadata{
+			alertName: alertName,
+			podName:   string(alert.Labels["pod"]),
+			namespace: string(alert.Labels["namespace"]),
+		}
+
+		// check if dead mans switch is firing
+		if alertName == deadMansSwitch && alert.State != prometheusv1.AlertStateFiring {
+			alertsError.deadMansSwitchFiring = false
+		}
+		// check for firing alerts
+		if alertName != deadMansSwitch {
+			if alert.State == prometheusv1.AlertStateFiring {
+				alertsFired = true
+				alertsError.alertsFiring = append(alertsError.alertsFiring, alertMetadata)
+
+			}
+
+		}
+	}
+
+	if alertsError.isValid() {
+		return alertsError
+
+	}
+	return nil
+}
+
+func podLogs(t *testing.T, ctx *TestingContext) {
+	if alertsFired == true {
```

Comment:
Since this will always be called if the test fails, I think we can get rid of this check